Userland/Libraries/LibGfx/PNGWriter.cpp

```diff
@@ -13,18 +14,32 @@
 namespace Gfx {
 
 class PNGChunk {
+    using data_length_type = u32;
+
 public:
     explicit PNGChunk(String);
-    Vector<u8> const& data() const { return m_data; };
+    auto const& data() const { return m_data; };
     String const& type() const { return m_type; };
-    void add_u8(u8);
-    void add_u16_big(u16);
-    void add_u32_big(u32);
-    void add_u16_little(u16);
-    void add_u32_little(u32);
+    void reserve(size_t bytes) { m_data.ensure_capacity(bytes); }
+
+    template<typename T>
+    void add_as_big_endian(T);
+
+    template<typename T>
+    void add_as_little_endian(T);
+
+    template<typename T>
+    void add(T);
```

Comment:
* maybe set some Restrictions on those types
* if you have 'safe' endian aware setters, exposing the raw setter might be redundant and could lead to unsafe code in the future, can you make it private?

Reply:
Added `requires(IsUnsigned<T>)` for add and made it private. Does that work?